File: app/src/main/java/org/neidhardt/dynamicsoundboard/dialog/BaseDialog.java
package org.neidhardt.dynamicsoundboard.dialog;

import android.app.DialogFragment;
import android.view.View;
import org.neidhardt.dynamicsoundboard.NavigationDrawerFragment;
import org.neidhardt.dynamicsoundboard.soundcontrol.SoundSheetFragment;
import org.neidhardt.dynamicsoundboard.soundmanagement.ServiceManagerFragment;
import org.neidhardt.dynamicsoundboard.soundsheet.SoundSheetsManagerFragment;

/**
 * File created by eric.neidhardt on 14.11.2014.
 */
public abstract class BaseDialog extends DialogFragment
{
	private View mainView;

	public View getMainView()
	{
		return mainView;
	}

	protected void setMainView(View mainView)
	{
		this.mainView = mainView;
	}

	public ServiceManagerFragment getServiceManagerFragment()
	{
		return (ServiceManagerFragment)this.getFragmentManager().findFragmentByTag(ServiceManagerFragment.TAG);
	}

	public SoundSheetsManagerFragment getSoundSheetManagerFragment()
	{
		return (SoundSheetsManagerFragment)this.getFragmentManager().findFragmentByTag(SoundSheetsManagerFragment.TAG);
	}

	public SoundSheetFragment getSoundSheetFragment(String fragmentTag)
	{
		return (SoundSheetFragment)this.getFragmentManager().findFragmentByTag(fragmentTag);
	}

	public NavigationDrawerFragment getNavigationDrawerFragment()
	{
		return (NavigationDrawerFragment)this.getFragmentManager().findFragmentByTag(NavigationDrawerFragment.TAG);
	}
}

File: app/src/main/kotlin/org/neidhardt/dynamicsoundboard/base/BaseDialog.kt
package org.neidhardt.dynamicsoundboard.base

import android.support.v4.app.DialogFragment
import org.neidhardt.dynamicsoundboard.soundactivity.SoundActivity
import org.neidhardt.dynamicsoundboard.splashactivity.SplashActivity

/**
 * File created by eric.neidhardt on 14.11.2014.
 */
abstract class BaseDialogFragment : DialogFragment()

abstract class BaseDialog : BaseDialogFragment() {
	companion object {
		val KEY_CALLING_FRAGMENT_TAG: String = "KEY_CALLING_FRAGMENT_TAG"
	}

	val soundActivity: SoundActivity
		get() = this.activity as SoundActivity

	val splashActivity: SplashActivity
		get() = this.activity as SplashActivity
}

